src/azure-cli/azure/cli/command_modules/appconfig/_kv_helpers.py

```diff
@@ -0,0 +1,720 @@
+        for keys, values in value_to_flatten.items():
+            current_prefix = prefix + separator + str(keys) if prefix else str(keys)
+            if isinstance(values, (dict, list)):
+                flattened_properties_dict.update(__flatten_dict_to_properties(values, separator, current_prefix))
+            else:
+                flattened_properties_dict.update({current_prefix: str(values)})
+
+    elif isinstance(value_to_flatten, list):
+        for idx, val in enumerate(value_to_flatten):
+            current_prefix = prefix + '[' + str(idx) + ']' if prefix else str(idx)
+            if isinstance(val, (dict, list)):
+                flattened_properties_dict.update(__flatten_dict_to_properties(val, separator, current_prefix))
+            else:
+                flattened_properties_dict.update({current_prefix: str(val)})
+    else:
+        flattened_properties_dict.update({prefix: str(value_to_flatten)})
+
+    return flattened_properties_dict
+
+
+def __export_features(retrieved_features, format_):
+    exported_dict = {}
+    client_filters = []
+    if format_ == 'json':
+        exported_dict["FeatureManagement"] = {}
+
+    elif format_ in ('yaml', 'properties'):
+        # Currently, this condition will never be met
+        # We only support json feature flags for now
+        featureSetValues = {}
+        featureSetValues["features"] = {}
+        exported_dict["feature-management"] = {"featureSet": featureSetValues}
+
+    try:
+        # retrieved_features is a list of FeatureFlag objects
+        for feature in retrieved_features:
+
+            # if feature state is on or off, it means there are no filters
+            if feature.state == "on":
+                feature_state = True
+
+            elif feature.state == "off":
+                feature_state = False
+
+            elif feature.state == "conditional":
+                feature_state = {}
+                feature_state["EnabledFor"] = []
+                client_filters = feature.conditions["client_filters"]
+                # client_filters is a list of dictionaries, where all dictionaries have 2 keys - Name and Parameters
+                for filter_ in client_filters:
+                    feature_filter = {}
+                    feature_filter["Name"] = filter_.name
+                    feature_filter["Parameters"] = filter_.parameters
+                    feature_state["EnabledFor"].append(feature_filter)
+
+            feature_entry = {}
+            feature_entry[feature.key] = feature_state
+
+            if format_ == 'json':
+                exported_dict["FeatureManagement"].update(feature_entry)
+
+            elif format_ in ('yaml', 'properties'):
+                featureSetValues["features"].update(feature_entry)
+                exported_dict["feature-management"].update({"featureSet": featureSetValues})
+
+        if format_ == 'properties':
+            exported_dict = __flatten_dict_to_properties(exported_dict)
```

Comment:
kind of confusing, for feature flags we only support json right? For properties or yaml, what is expected behavior?

Reply:
Logging warnings and ignoring feature flags while exporting to yaml/properties file.

---
In reply to: [356334456](https://github.com/Azure/azure-cli/pull/11467#discussion_r356334456) [](ancestors = 356334456)